feat: Support configurable timeouts in cloud_backup_snapshot (#3171)
mongodb/terraform-provider-mongodbatlas: docs/resources/cloud_backup_snapshot.md

* `cluster_name` - (Required) The name of the Atlas cluster that contains the snapshots you want to retrieve.
* `description` - (Required) Description of the on-demand snapshot.
* `retention_in_days` - (Required) The number of days that Atlas should retain the on-demand snapshot. Must be at least 1.

> * `timeouts`- (Optional) The duration of time to wait for a Cloud Backup Snapshot to be created. The timeout value is defined by a signed sequence of decimal numbers with an time unit suffix such as: `1h45m`, `300s`, `10m`, .... The valid time units are:  `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`. The default timeout for create is `1h`. Learn more about timeouts [here](https://www.terraform.io/plugin/sdkv2/resources/retries-and-customizable-timeouts).

Reviewer: Is `....` correct?

> * `timeouts`- (Optional) The duration of time to wait for a Cloud Backup Snapshot to be created. The timeout value is defined by a signed sequence of decimal numbers with an time unit suffix such as: `1h45m`, `300s`, `10m`, .... The valid time units are:  `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`. The default timeout for create is `1h`. Learn more about timeouts [here](https://www.terraform.io/plugin/sdkv2/resources/retries-and-customizable-timeouts).

Reviewer 2: ```suggestion
* `timeouts`- (Optional) The duration of time to wait for Atlas to create a Cloud Backup Snapshot. The timeout value is defined by a signed sequence of decimal numbers with a time unit suffix such as: `1h45m`, `300s`, `10m`, etc. The valid time units are:  `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`. Defaults to `1h`. Learn more about timeouts [here](https://www.terraform.io/plugin/sdkv2/resources/retries-and-customizable-timeouts).
```

Author: Applied suggestion to all our other resources docs as well for consistency.
